apps/mobile/src/features/ImportSigner/hooks/useImportPrivateKey.ts

```diff
@@ -0,0 +1,115 @@
+import { ethers } from 'ethers'
+import { useState } from 'react'
+import Clipboard from '@react-native-clipboard/clipboard'
+import { useLocalSearchParams, useRouter } from 'expo-router'
+import { storePrivateKey } from '@/src/hooks/useSign/useSign'
+import useDelegate from '@/src/hooks/useDelegate'
+import Logger from '@/src/utils/logger'
+import { detectInputType, InputType } from '@/src/utils/inputDetection'
+
+const ERROR_MESSAGE = 'Invalid private key or seed phrase.'
+export const useImportPrivateKey = () => {
+  const [input, setInput] = useState('')
+  const [inputType, setInputType] = useState<InputType>('unknown')
+  const [wallet, setWallet] = useState<ethers.Wallet>()
+  const local = useLocalSearchParams<{ safeAddress: string; chainId: string; import_safe: string }>()
+  const [error, setError] = useState<string | undefined>(undefined)
+  const router = useRouter()
+  const { createDelegate } = useDelegate()
+
+  const handleInputChange = (text: string) => {
+    setInput(text)
+    const detectedType = detectInputType(text)
```

Comment:
if we could add the detectedType into an enum would be nice, so we would avoid having these arbitrary strings comparisons spread into the code

Reply:
I don't undrestand why you guys are always so into enums 🤷 

The return type of detectInputType is 

export type InputType = 'private-key' | 'seed-phrase' | 'unknown'

I know that `if (detectedType === 'private-key')` looks like a string comparison and it is, but typescript will know if something is wrong. I also don't see the benefit of yet importing another enum on top just to do

`detectedType === detectInputTypes.privateKey` - to me this is way more work than just comparing against `private-key`? 

If we ever come to change `private-key` to `some-other-fancy-name-for-private-key`, then renaming with an IDE is still like 1s job 🤷 

Honest question. What is the benefit of enums?

Reply:
for me it is not the enum its self, but having to dealing with these arbitrary values everywhere. 🤔